### DISCUSSION TITLE
Text align and arrows on small (value) dimenions are not right

### DISCUSSION THREAD

**Discussion by Jopie01** [mozman/ezdxf#1207] [Q&A] [answered]
I have some trouble with dimensions which have small values. I have created a DXF file which sets the dimension styles and use that as a template. When I add dimensions in paperspace and do a transform because of the scale of the view, I end up with a very strange dimension. I opened the exported DXF file in Bricscad.

![Screenshot_20241122_113719](https://github.com/user-attachments/assets/f2f388c6-59fc-4d7d-a26f-6105c4a13093)

When I select the dimension and change for example the layer, it will automatically update to the correct settings.

![Screenshot_20241122_114907](https://github.com/user-attachments/assets/488dcdd5-497e-49d7-86db-d859444fdf78)

Below the function I'm using which adds the dimension. Previously I transformed the whole dimension, but rewrote it to only transform the needed points.

```python
def add_dimension(self, layout, viewport, start, end, dim_type, offset=20, text=None):
    # get the scale and transformation matrix
    matrix = viewport.get_transformation_matrix()
    scale = viewport.get_scale()

    start_pos = matrix.transform(start)
    end_pos = matrix.transform(end)

    if dim_type == 'horizontal':
        base = (0, offset)
        angle = 0
    else:
        base = (offset, 0)
        angle = 90

    dim_place = matrix.transform(base)

    dim = layout.add_linear_dim(
        base=dim_place,
        angle=angle,
        p1=start_pos,
        p2=end_pos,
        dimstyle='ISO-25',
        override={
            'dimlfac': 1 / scale,
            'dimdec': 0,
        }
    )
    if text:
        dim.set_text_format(postfix=text)

    dim.render()
```
This works perfectly for placing the dimensions and for the larger dimensions. But the smaller ones needs some extra attention.  Do I need an extra check myself for small values? Or is this something in ezdxf I can use?

**Comment by mozman** [comment, marked as answer]
The `ezdxf` dimension renderer will never be on par with CAD applications!

Here is a tutorial for overriding the default text location: https://ezdxf.mozman.at/docs/tutorials/linear_dimension.html#user-defined-text-locations

**Comment by Jopie01** [comment]
> The ezdxf dimension renderer will never be on par with CAD applications!

That would be impossible and wasn't my question. I hoped if there was a kind of function to give me the result if the arrows and text were fitting between the lines. Then I could take action on that. For example to flip the arrows, I swap the two points. For shifting the text I'm using https://ezdxf.readthedocs.io/en/stable/tutorials/linear_dimension.html#location-relative-to-default-location
